internal/deleter/delete.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2022 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deleter
+
+import (
+	"context"
+
+	"github.com/palantir/log4j-sniffer/pkg/crawl"
+	"github.com/palantir/log4j-sniffer/pkg/log"
+)
+
+// Deleter will delete files containing vulnerable findings given they match certain criteria.
+type Deleter struct {
+	Logger        log.Logger
+	FilepathMatch func(filepath string) (bool, error)
+	FindingMatch  func(finding crawl.Finding) bool
+	DryRun        bool
+	Delete        func(filepath string) error
+}
+
+// Process a finding and delete it if it is eligible for deletion given certain configuration criteria.
+//
+// If the filepath and detailed finding both match for a given crawl.Path then a file is eligible for deletion.
+// In this case, Process will always return false to state that this file should no longer exist and that inspecting
+// this file for more findings should  not be undertaken.
+//
+// When Deleter.DryRun is true then a line will be logged stating that the file would be deleted.
+// When Deleter.Delete is false, then the configured function Delete will be called to delete the file.
+func (d Deleter) Process(ctx context.Context, path crawl.Path, finding crawl.Finding, _ crawl.Versions) bool {
+	if len(path) == 0 {
+		return true
+	}
+	match, err := d.FilepathMatch(path[0])
```

Comment:
This will currently crash if `d.FilepathMatch` is not specified (since default value is `nil`). Generally, it is best form for exported structs to be functional/have sensible defaults in their empty form.

Main options would be:
1. Define behavior when these values aren't specified (match all/match none)
2. Check if these values aren't specified and throw an appropriate error on function invocation
3. Unexport the struct, make it an interface and then have creation function return properly initialized one